client/client.go

```diff
@@ -269,3 +271,16 @@ func GetExponentialBackoffTime(tried int, maxInterval time.Duration) (time.Durat
 
 	return interval, nil
 }
+
+// unMarshalErrorMessage unmarshals the error message contained in an
+// error request from the server.
+func unMarshalErrorMessage(r io.Reader) string {
+	e := new(struct {
+		Error string `json:"error"`
+		ReqID string `json:"request_id"`
+	})
+	if err := json.NewDecoder(r).Decode(e); err != nil {
+		return fmt.Sprintf("failed to parse server response: %v", err)
+	}
+	return fmt.Sprintf("request failed with error: %s : request_id: %s", e.Error, e.ReqID)
```

Comment:
Maybe you can change the log message to something like `request (id: %s) error: %s` since this will be a part of the "failed to upload the client logs: error-code: %d : %s" message which is indicating the error anyway.